### PR TITLE
Fixes tzname return type

### LINKED ISSUE #7630: Unicode issue in matplotlib.dates
The import of `__future__.unicode_literals` in `matplotlib.dates` causes an error when calling `datetime.datetime.tzname()` on a datetime created with `tzinfo=matplotlib.dates.UTC`:
```
>>> import matplotlib.dates
>>> import datetime
>>> dt = datetime.datetime(2000,1,1, tzinfo=matplotlib.dates.UTC)
>>> dt.tzname()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: tzinfo.tzname() must return None or a string, not 'unicode'
```
This causes issues when plotting with a date formatter:

```
>>> import matplotlib.pyplot as plt
>>> import matplotlib.dates
>>> plt.plot([700000], [0])
[<matplotlib.lines.Line2D object at 0x7fe41dbb4050>]
>>> plt.gca().xaxis.set_major_formatter(matplotlib.dates.DateFormatter('%Z'))
>>> plt.show()
...
  File .../lib/python2.7/site-packages/matplotlib/dates.py", line 589, in strftime
    return cbook.unicode_safe(dt.strftime(fmt))
TypeError: tzinfo.tzname() must return None or a string, not 'unicode'
```
(I've omitted most of the call stack)

Matplotlib 1.5.3, Python 2.7.12, Linux
Installed from source
The issue doesn't exist in Matplotlib 1.3.1

### PR DESCRIPTION
**Pull request by Kojoley**
Closes #7630

### CODE REVIEW

File: lib/matplotlib/tests/test_dates.py [hunk 1]
```diff
@@ -465,6 +465,11 @@ def test_DayLocator():
    mdates.DayLocator(interval=1.0)
 
 
+def test_tz_utc():
+    dt = datetime.datetime(1970, 1, 1, tzinfo=mdates.UTC)
+    dt.tzname()
```
Comment (dopplershift) [inline lib/matplotlib/tests/test_dates.py]: Why not check the return value of `dt.tzname()` rather than making this just a smoke test?

Comment (dopplershift) [inline lib/matplotlib/tests/test_dates.py]: I should add that my big complaint here is that this test, while exercising the method, does nothing to actually catch the original problem.

Comment (tacaswell) [inline lib/matplotlib/tests/test_dates.py]: This does, the exception is coming out of datetime

```python
Python 2.7.13 |Continuum Analytics, Inc.| (default, Dec 20 2016, 23:09:15)                                                                                                                                                                                                                                                                                                                                                              
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2                                                                                                                                                                                                                                                                                                                                                                                        
Type "help", "copyright", "credits" or "license" for more information.                                                                                                                                                                                                                                                                                                                                                                  
Anaconda is brought to you by Continuum Analytics.                                                                                                                                                                                                                                                                                                                                                                                      
Please check out: http://continuum.io/thanks and https://anaconda.org                                                                                                                                                                                                                                                                                                                                                                   
>>> import matplotlib.dates as mdates                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
>>> import datetime                                                                                                                                                                                                                                                                                                                                                                                                                     
>>> dt = datetime.datetime(1970, 1, 1, tzinfo=mdates.UTC)                                                                                                                                                                                                                                                                                                                                                                               
>>> dt.tzname()                                                                                                                                                                                                                                                                                                                                                                                                                         
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                                                                      
  File "<stdin>", line 1, in <module>                                                                                                                                                                                                                                                                                                                                                                                                   
TypeError: tzinfo.tzname() must return None or a string, not 'unicode'                                                                                                                                                                                                                                                                                                                                                                  
>>>                                                                                                                                                                                                                                                                                                                                                                                                                                     
        
```

Comment (dopplershift) [inline lib/matplotlib/tests/test_dates.py]: AH. I see now. My mistake.

**Comment by tacaswell** [comment]
Thanks! 

Backported to v2.x as c602d8269

**Pull request merged**